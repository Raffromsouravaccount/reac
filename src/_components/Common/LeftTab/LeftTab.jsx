import React from "react";
import Tabs from "@material-ui/core/Tabs";
import Tab from "@material-ui/core/Tab";
import RadioButtonUncheckedIcon from "@material-ui/icons/RadioButtonUnchecked";
import RadioButtonCheckedIcon from "@material-ui/icons/RadioButtonChecked";
import CheckCircleIcon from "@material-ui/icons/CheckCircle";

const LeftTab = ({
  className,
  name,
  options,
  formFilled,
  selectedTab,
  handleChange,
}) => (
  <Tabs
    className={className}
    orientation="vertical"
    variant="scrollable"
    value={selectedTab}
    onChange={(event, newValue) => handleChange(event, newValue)}
  >
    {options.map((option, index) => (
      <Tab
        key={index}
        className={option.status === true ? "completed" : ""}
        label={option.label}
        icon={
          option.status ? (
            <CheckCircleIcon />
          ) : selectedTab === 0 ? (
            <RadioButtonCheckedIcon />
          ) : (
            <RadioButtonUncheckedIcon />
          )
        }
      />
    ))}
  </Tabs>
);
export default LeftTab;
