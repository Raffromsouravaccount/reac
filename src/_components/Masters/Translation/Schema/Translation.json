[
  {
    "name": "ar",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Arabic",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "as",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Assamese",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "id",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Bahasa (Indonesia)",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "bn",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Bengali",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "hr",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Bhojpuri",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "fr",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "French",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "de",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "German",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "gu",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Gujarati",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "hi",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Hindi",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "km",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Kannada",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "ko",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Korean",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "ms",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Malay",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "ml",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Malayalam",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "zh",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Mandarin",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "mr",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Marathi",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "or",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Odia",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "pa",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Punjabi",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "ru",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Russian",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "es",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Spanish",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "ta",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Tamil",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "te",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Telugu",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  },
  {
    "name": "th",
    "value": "",
    "col": "col-md-6 col-lg-4",
    "type": "text",
    "moreText": "Thai",
    "errorText": "",
    "validation": {
      "maxLength" : 100
    }
  }
]
