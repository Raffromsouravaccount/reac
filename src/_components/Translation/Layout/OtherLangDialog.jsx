import React, { Component } from "react";
import Button from "@material-ui/core/Button";
import Dialog from "@material-ui/core/Dialog";
import DialogActions from "@material-ui/core/DialogActions";
import DialogContent from "@material-ui/core/DialogContent";
import DialogTitle from "@material-ui/core/DialogTitle";

// import './Model.css';
import { constantText } from "../../../_helpers/constants.text";
import { connect } from "react-redux";
import CheckBox from "../../Common/CheckBox/CheckBox";
import Config from "../../../Config/config";

//svg
import TickIcon from "images/tick.svg";

class OtherLangDialog extends Component {
  constructor(props) {
    super(props);
    this.state = {
      languageList: [],
    };
  }

  componentDidMount() {
    const languageList = JSON.parse(JSON.stringify(this.props.languageList));
    const filterList = languageList.length && languageList.filter((item) => item.code != Config.defaultLanguageCode) || [];
    const assignLang = [...this.props.languages];
    const myArrayFiltered  = filterList.map((el) => {
      const temp = assignLang.find((f) => f.code === el.code);
      if(temp) {
        el['display'] = false; 
      } else {
        el['display'] = true; 
      }
      return el;
    });
    this.setState({ languageList: myArrayFiltered });
  }

  onLanguageSelectionFromDialog = (i) => {
    const languageList = this.state.languageList;
    const currentState = languageList[i]["selectedFromDialog"] || false;
    languageList[i]["selectedFromDialog"] = currentState ? !currentState : true;
    languageList[i]["label"] = languageList[i]["title"];
    this.setState({ languageList });
  };

  render() {
    let { state, className, btn1Action, btn2Action, handleClose } = this.props;
    const { languageList } = this.state;
    return (
      <Dialog open={state} onClose={handleClose} className={className}>
        {
          <DialogTitle>
            {constantText.translations.other_lang_text}
          </DialogTitle>
        }

        <DialogContent>
          <div className="row">
            {languageList &&
              languageList.map((language, index) => (
                language.display && <div key={index} className="col-6 col-md-4 col-lg-3 lang-name">
                  <CheckBox
                    key={index}
                    className={`zee-checkbox-field auto-${language?.title ? language?.title.split(" ").join(""):""}-${index}`}
                    label={language?.title}
                    // name={key}
                    handleCheckBox={(e) =>
                      this.onLanguageSelectionFromDialog(index)
                    }
                    // disabled={isDisable}
                    // labelPlacement={form?.labelPlacement}
                    checked={languageList[index].selectedFromDialog || false}
                    required={false}
                  />
                  <div className="trans-status">
                    <span className="inner">
                      { language && (String(language?.translationStatus) && language?.translationStatus == "2") && 
                        <span className="completed">
                          &#x25CF;&nbsp; {constantText.translations.completed_text}
                          <span className="mark-green flex align-items-center justify-content-center">
                            <TickIcon />
                          </span>
                        </span>
                      }
                      { language && (String(language?.translationStatus) && language?.translationStatus == "1") && <span className="partiallyDone">&#x25CF;&nbsp; {constantText.translations.partially_done_text}</span>}
                      { language && (!language?.translationStatus || language?.translationStatus == "0") && <span className="noTranslation">&#x25CF;&nbsp; {constantText.translations.no_translation_text}</span>}
                    </span>
                  </div>
                </div>
              ))}
          </div>
        </DialogContent>
        <DialogActions>
          {
            <Button className="auto-button-Yes" onClick={() => btn1Action(languageList)}>
              {constantText.translations.other_lang_popup_btn_1_text}
            </Button>
          }
          {
            <Button className="auto-button-No" onClick={btn2Action}>
              {constantText.translations.other_lang_popup_btn_2_text}
            </Button>
          }
        </DialogActions>
      </Dialog>
    );
  }
}

export default connect(null, {})(OtherLangDialog);
