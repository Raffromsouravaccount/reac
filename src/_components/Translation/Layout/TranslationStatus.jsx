import React from "react";

//svg
import TickIcon from "images/tick.svg";
import Config from "../../../Config/config";
import { constantText } from "../../../_helpers/constants.text";

const TranslationStatus = ({ languageList, onClick, handleLanguageIcon }) => {
  const filterList = languageList.length && languageList.filter((item) => item.code != Config.defaultLanguageCode) || [];
  return (
    <div className="row">
      {filterList && filterList.map((language, index) => 
    
          { return language && <div key={index} className="col-sm-6 col-md-4 col-lg-3 col-xl-2">
            <div className={`listBox lang-list auto-${language?.title ? language?.title.split(" ").join(""):""}-${index}`} onClick={() => onClick(language)}>
            { language && (language?.isDone) &&
              <span className="mark-done-circle flex align-items-center">
                <span className="mark-green flex align-items-center justify-content-center">
                  <TickIcon />
                </span>
                Mark as Done
              </span>
            }
              <div className="icon">
                <img src={language && language?.image ? language?.image : handleLanguageIcon(language)} />
              </div>
              <div className="lang-name">{language?.title}</div>
              { language && (String(language?.translationStatus) && language?.translationStatus == "2") && <div className="lang-status completed">&#x25CF;&nbsp; {constantText.translations.completed_text}</div> }
              { language && (String(language?.translationStatus) && language?.translationStatus == "1") && <span className="lang-status partiallyDone">&#x25CF;&nbsp; {constantText.translations.partially_done_text}</span>}
              { language && (!language?.translationStatus || language?.translationStatus == "0") && <span className="lang-status noTranslation">&#x25CF;&nbsp; {constantText.translations.no_translation_text}</span>}
            </div>
          </div> }
        )}
    </div>
  );
};

export default TranslationStatus;
