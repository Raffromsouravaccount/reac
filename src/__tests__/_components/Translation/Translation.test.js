import React from 'react';
import { shallow, mount } from 'enzyme';
import { storeFactory } from '../../../Utils';

import Translation from './../../../_components/Translation/Translation';
import TranslationStatus from './../../../_components/Translation/Layout/TranslationStatus';
import OtherLangDialog from './../../../_components/Translation/Layout/OtherLangDialog';
import TranslationMovie from '../../../_components/Translation/TranslationMovie/TranslationMovie';
import LeftTab from './../../../_components/Common/LeftTab/CommonLeftTab';
import { CommonModel } from './../../../_components/Common/Model/CommonModel';
import TranslationCastProfile from './../../../_components/Translation/TranslationCastProfile/TranslationCastProfile';
import TranslationCollection from './../../../_components/Translation/TranslationCollection/TranslationCollection';
import TranslationTvshow from './../../../_components/Translation/TranslationTvshow/TranslationTvshow';
import { constantText } from './../../../_helpers/constants.text';
import TranslationVideo from './../../../_components/Translation/TranslationVideo/TranslationVideo';
import { translationService } from './../../../_services/translation.service';
import TranslationSeason from './../../../_components/Translation/TranslationSeason/TranslationSeason';
import TranslationEpisode from './../../../_components/Translation/TranslationEpisode/TranslationEpisode';


const setup = (initialState, props = {...matchProps}, state = null) => {
    const store = storeFactory(initialState);
    const wrapper = shallow(<Translation store={store} {...props} />).dive();
    if (state) wrapper.setState(state);
    return wrapper;
}

const initialState = {
    user_reducer: {
        userdetails: {
            translationLanguages: [
                {
                  "id": "7c59f385-57a4-4a04-ac7b-269ea2460abc",
                  "title": "Hindi",
                  "code": "hi"
                },
                {
                  "id": "eb4c9b92-0a9f-44fd-b4dd-f9e8f5628726",
                  "title": "French",
                  "code": "fr"
                }
            ],
        }
    }
};

const matchProps = {
    match:{
        params:{
            movieId: "5753e295-66e8-4a42-9f90-af14c3edadab"
        },
        url: "/movie/edit/5753e295-66e8-4a42-9f90-af14c3edadab/translation",
        path: "/movie/view/:movieId/translation",
    },
    location: {
        state: {
            language: [{title: 'Chinese', code: 'cr'}]
        }
    }
};

describe('<Translation />', () => {
    let wrapper;
    const userData = {
        userID: "d2fa1282-5f91-4231-a2b4-f308caf5afc6"
    };
    beforeEach(() => {
        wrapper = setup();
    });
    it('Should renders Translation default', () => {
        expect(wrapper.exists()).toBe(true);
    });
    it('Should get userID from localStorage', () => {
        Storage.prototype.getItem = jest.fn(() => userData.userID);
        const result = localStorage.getItem();
        expect(result).toEqual(userData.userID);
    });
    it('Sholud test componentDidMount', () => {
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        jest.spyOn(instance, 'componentDidMount');
        instance.componentDidMount();
        expect(instance.componentDidMount).toHaveBeenCalledTimes(1);
    });
    it('Sholud test componentWillReceiveProps', () => {
        const nextProps = {
            userDetails: {...initialState.user_reducer.userdetails}
        };
        const userLanguages = nextProps.userDetails.translationLanguages;
        const translationPage = constantText.translations.pages.movie_text;
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        jest.spyOn(instance, 'componentWillReceiveProps');
        instance.componentWillReceiveProps(nextProps);
        expect(instance.componentWillReceiveProps).toHaveBeenCalledTimes(1);
        expect(nextProps.userDetails.translationLanguages).toBeTruthy();
        expect(translationService.permissionHandler(translationPage)?.edit()).toBeFalsy();
        expect(translationService.permissionHandler(translationPage)?.view()).toBeFalsy();
        expect(userLanguages.length > 0).toBeTruthy();
    });
    it('Sholud test onLangDialogClose', () => {
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        jest.spyOn(instance, 'onLangDialogClose');
        instance.onLangDialogClose();
        expect(instance.onLangDialogClose).toHaveBeenCalled();
    });
    it('Sholud test onLangDialogClose with langDialogSubmit', () => {
        const langDialogSubmit = true;
        const showLangDialog = true;
        const selectedLangTab = 0;
        const languageList = [{title: 'German', code: 'gr', selectedFromDialog: true}];
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        testComp.setState({selectedLangTab});
        jest.spyOn(instance, 'onLangDialogClose');
        instance.onLangDialogClose(showLangDialog, langDialogSubmit, languageList);
        expect(instance.onLangDialogClose).toHaveBeenCalled();
    });
    it('Sholud test other-lang-btn onClick', () => {
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        const spy = jest.spyOn(instance, 'onLangDialogClose');
        instance.forceUpdate();
        testComp.update();
        const button = testComp.find('.other-lang-btn');
        button.simulate('click');
        expect(spy).toHaveBeenCalled();
    });
    it('Sholud test handleRoute', () => {
        const showLanguageList = true;
        const permissionAddEdit = true;
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        testComp.setState({showLanguageList, permissionAddEdit});
        jest.spyOn(instance, 'handleRoute');
        instance.handleRoute();
        expect(instance.handleRoute).toHaveBeenCalled();
        expect(testComp.state('permissionAddEdit')).toBeTruthy();
        expect(testComp.state('showLanguageList')).toBeTruthy();
    });
    it('Sholud test handleRoute onClick', () => {
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        const spy = jest.spyOn(instance, 'handleRoute');
        instance.forceUpdate();
        testComp.update();
        const button = testComp.find('.back-user-btn > span');
        button.simulate('click');
        expect(spy).toHaveBeenCalled();
    });
    it('Sholud test handleLangTab', () => {
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        const selectedLangTab = testComp.state('selectedLangTab');
        jest.spyOn(instance, 'handleLangTab');
        instance.handleLangTab(selectedLangTab);
        expect(instance.handleLangTab).toHaveBeenCalled();
    });
    it('Sholud test onNavToAssignedLang', () => {
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        jest.spyOn(instance, 'onNavToAssignedLang');
        instance.onNavToAssignedLang();
        expect(instance.onNavToAssignedLang).toHaveBeenCalled();
    });
    it('Sholud test onNavToAssignedLang onClick', () => {
        const userLanguages = [{title: 'Hindi', code: 'hi'}];
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        testComp.setState({showNavToAssignedLang: true, userLanguages});
        const spy = jest.spyOn(instance, 'onNavToAssignedLang');
        instance.forceUpdate();
        testComp.update();
        expect(testComp.state('showNavToAssignedLang')).toBeTruthy();
        expect(testComp.state('userLanguages')?.length).toBeGreaterThanOrEqual(0);
        const button = testComp.find('.back-arrow-assign-lang');
        button.simulate('click');
        expect(spy).toHaveBeenCalled();
    });
    it('Sholud render getAssignedLanguages', () => {
        const showLanguageList = false;
        const userLanguages = [{title: 'Hindi', code: 'hi'}];
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        jest.spyOn(instance, 'getAssignedLanguages');
        testComp.setState({showLanguageList, userLanguages});
        expect(testComp.state('showLanguageList')).toBeFalsy();
        expect(testComp.state('userLanguages')).toHaveLength(1);
        instance.getAssignedLanguages();
        expect(instance.getAssignedLanguages).toHaveBeenCalled();
    });
    it('Sholud test onLanguageSelectionFromStatusView', () => {
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        jest.spyOn(instance, 'onLanguageSelectionFromStatusView');
        instance.onLanguageSelectionFromStatusView();
        expect(instance.onLanguageSelectionFromStatusView).toHaveBeenCalled();
    });
    it('Sholud test updateMarkAsDoneAction', () => {
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        jest.spyOn(instance, 'updateMarkAsDoneAction');
        instance.updateMarkAsDoneAction();
        expect(instance.updateMarkAsDoneAction).toHaveBeenCalled();
    });
    it('Sholud test getLangIcon', () => {
        const langIcons = constantText.langIcons;
        const output = setup();
        const testComp = output.dive();
        const instance = testComp.instance();
        testComp.setState({langIcons});
        jest.spyOn(instance, 'getLangIcon');
        const lang = {title: 'Hindi'};
        instance.getLangIcon(lang);
        expect(instance.getLangIcon).toHaveBeenCalledWith(lang);
    });
    it('Sholud render Common Model', () => {
        const markAsDoneDialogShow = true;
        const output = setup();
        const testComp = output.dive();
        testComp.setState({markAsDoneDialogShow});
        expect(testComp.state('markAsDoneDialogShow')).toBeTruthy();
        expect(testComp.containsMatchingElement(<CommonModel />)).toBe(true);
    });
    it('Should renders Translation Status', () => {
        const showLanguageList = true;
        const permissionViewOnly = true;
        const output = setup();
        const testComp = output.dive();
        testComp.setState({showLanguageList, permissionViewOnly})
        expect(testComp.state('showLanguageList')).toBeTruthy();
        expect(testComp.state('permissionViewOnly')).toBeTruthy();
        expect(testComp.containsMatchingElement(<TranslationStatus />)).toBe(true);
    });
    it('Should renders OtherLangDialog', () => {
        const showLangDialog = true;
        const output = setup();
        const testComp = output.dive();
        testComp.setState({showLangDialog})
        expect(testComp.state('showLangDialog')).toBeTruthy();
        expect(testComp.containsMatchingElement(<OtherLangDialog />)).toBe(true);
    });
    it('Should renders LeftTab', () => {
        const showLanguageList = false;
        const output = setup();
        const testComp = output.dive();
        testComp.setState({showLanguageList});
        expect(testComp.state('showLanguageList')).toBeFalsy();
        expect(testComp.containsMatchingElement(<LeftTab />)).toBe(true);
    });
    it('Should renders TranslationMovie Component', () => {
        const translationPage = constantText.translations.pages.movie_text;
        const languages = ['hi'];
        const showLanguageList = false;
        const output = setup();
        const testComp = output.dive();
        testComp.setState({translationPage, languages, showLanguageList});
        expect(testComp.state('showLanguageList')).toBeFalsy();
        expect(testComp.state('translationPage')).toEqual('movie');
        expect(testComp.state('languages')).toHaveLength(1);
        expect(testComp.containsMatchingElement(<TranslationMovie />)).toBe(true);
    });
    it('Should renders TranslationCastProfile Component', () => {
        const translationPage = constantText.translations.pages.cast_text;
        const languages = ['hi'];
        const showLanguageList = false;
        const output = setup();
        const testComp = output.dive();
        testComp.setState({translationPage, languages, showLanguageList});
        expect(testComp.state('showLanguageList')).toBeFalsy();
        expect(testComp.state('translationPage')).toEqual('cast');
        expect(testComp.state('languages')).toHaveLength(1);
        expect(testComp.containsMatchingElement(<TranslationCastProfile />)).toBe(true);
    });
    it('Should renders TranslationCollection Component', () => {
        const translationPage = constantText.translations.pages.collection_text;
        const languages = ['hi'];
        const showLanguageList = false;
        const output = setup();
        const testComp = output.dive();
        testComp.setState({translationPage, languages, showLanguageList});
        expect(testComp.state('showLanguageList')).toBeFalsy();
        expect(testComp.state('translationPage')).toEqual('collection');
        expect(testComp.state('languages')).toHaveLength(1);
        expect(testComp.containsMatchingElement(<TranslationCollection />)).toBe(true);
    });
    it('Should renders TranslationTvshow Component', () => {
        const translationPage = constantText.translations.pages.tvshow_text;
        const languages = ['hi'];
        const showLanguageList = false;
        const output = setup();
        const testComp = output.dive();
        testComp.setState({translationPage, languages, showLanguageList});
        expect(testComp.state('showLanguageList')).toBeFalsy();
        expect(testComp.state('translationPage')).toEqual('tvshow');
        expect(testComp.state('languages')).toHaveLength(1);
        expect(testComp.containsMatchingElement(<TranslationTvshow />)).toBe(true);
    });
    it('Should renders TranslationVideo Component', () => {
        const translationPage = constantText.translations.pages.video_text;
        const languages = ['hi'];
        const showLanguageList = false;
        const output = setup();
        const testComp = output.dive();
        testComp.setState({translationPage, languages, showLanguageList});
        expect(testComp.state('showLanguageList')).toBeFalsy();
        expect(testComp.state('translationPage')).toEqual('video');
        expect(testComp.state('languages')).toHaveLength(1);
        expect(testComp.containsMatchingElement(<TranslationVideo />)).toBe(true);
    });
    it('Should renders TranslationSeason Component', () => {
        const translationPage = constantText.translations.pages.season_text;
        const languages = ['hi'];
        const showLanguageList = false;
        const output = setup();
        const testComp = output.dive();
        testComp.setState({translationPage, languages, showLanguageList});
        expect(testComp.state('showLanguageList')).toBeFalsy();
        expect(testComp.state('translationPage')).toEqual('season');
        expect(testComp.state('languages')).toHaveLength(1);
        expect(testComp.containsMatchingElement(<TranslationSeason />)).toBe(true);
    });
    it('Should renders TranslationEpisode Component', () => {
        const translationPage = constantText.translations.pages.episode_text;
        const languages = ['hi'];
        const showLanguageList = false;
        const output = setup();
        const testComp = output.dive();
        testComp.setState({translationPage, languages, showLanguageList});
        expect(testComp.state('showLanguageList')).toBeFalsy();
        expect(testComp.state('translationPage')).toEqual('episode');
        expect(testComp.state('languages')).toHaveLength(1);
        expect(testComp.containsMatchingElement(<TranslationEpisode />)).toBe(true);
    });
});
