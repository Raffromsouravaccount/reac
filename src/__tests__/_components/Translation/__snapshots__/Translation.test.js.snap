// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Translation /> Should renders Translation component correctly 1`] = `
<div
  className="d-wrap c-n"
>
  <BreadcrumbsComp
    links={
      Array [
        Object {
          "color": "inherit",
          "route": "/dashboard",
          "text": "Dashboard",
        },
      ]
    }
    typography={
      Array [
        Object {
          "color": "textPrimary",
          "label": "primary",
          "text": "Translations",
        },
      ]
    }
  />
  <div
    className="profile-head flex align-items-center justify-content-between"
  >
    <div
      className="back-user-btn"
    >
      <span
        onClick={[Function]}
      >
        <test-file-stub />
      </span>
      <strong>
        <span>
          Translations
        </span>
      </strong>
    </div>
  </div>
  <div
    className="translation-wrapper"
  >
    <div
      className="whitebox lang-select-wrap m-b-30"
    >
      <div
        className="flex align-items-center justify-content-between"
      >
        <div
          className="left-area flex align-items-center"
        >
          <div
            className="lang-text"
          >
            Select Language
          </div>
          <LeftTab
            className="tabs lang-tab"
            handleChange={[Function]}
            options={Array []}
            orientation="horizontal"
            selectedTab={0}
            showIcon={false}
            variant="scrollable"
          />
        </div>
        <div
          className="right-area other-lang-btn"
          onClick={[Function]}
        >
          Other Languages
        </div>
      </div>
    </div>
  </div>
</div>
`;
